### PR TITLE
SNHUT 1244 & SNHUT 587 - Create Group - Frontend , Add users to groups via create group modal

### PR DESCRIPTION
**Pull request by sada-ajith**
Following tickets are addressed in this PR by @sada-ajith  and @shafi-sadaindia 

1. Group Management | Create Group - Frontend  - https://collectiveshift.atlassian.net/browse/SNHUT-587
2. Group Management | Add users to groups via create group modal - Frontend https://collectiveshift.atlassian.net/browse/SNHUT-1244

As of now API integration for both of these items are pending, will be handled later. 
Necessary unit tests are added
Please see screenshots as well
Screenshot 2022-08-02 at 2 18 26 PM
Screenshot 2022-08-02 at 2 18 35 PM
Screenshot 2022-08-02 at 2 20 43 PM
Screenshot 2022-08-02 at 2 21 08 PM

### CODE REVIEW

File: apps/toboggan-api/src/app/app.controller.ts [hunk 1]
```diff
@@ -65,4 +65,9 @@ export class AppController {
   deleteGroup(@Param('id') id) {
     return this.appService.deleteGroup(id);
   }
+
+  @Post('addusertogroup')
```
Comment (joao-sd) [inline apps/toboggan-api/src/app/app.controller.ts:69]: Does this follow the REST API endpoint pattern?

Maybe a post to /groups/user ?

File: apps/toboggan-app/src/app/group/components/add-users/add-users.component.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+    if (this.addUserForm.valid) {
+      this.groupService
+        .addUsertoGroup(
+          this.addUserForm.value.groupId,
+          this.addUserForm.value.user
+        )
+        .subscribe({
+          next: (response) => {
+            // handle success
```
Comment (joao-sd) [inline apps/toboggan-app/src/app/group/components/add-users/add-users.component.ts:50]: Should we remove these empty cases?

File: apps/toboggan-app/src/app/group/components/create-group/create-group.component.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  createGroup() {
+    this.createGroupForm.markAllAsTouched();
+    if (this.createGroupForm.valid) {
+      const group: INewGroup = {
+        name: this.createGroupForm.value.name,
+        description: this.createGroupForm.value.description,
+      };
+      this.groupService.createGroup(group).subscribe({
+        next: (response) => {
```
Comment (joao-sd) [inline apps/toboggan-app/src/app/group/components/create-group/create-group.component.ts:69]: Same comment as above

File: apps/toboggan-app/src/app/group/group.module.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    FormsModule,
+    StoriesModule,
+    ReactiveFormsModule, 
+    TypeaheadModule.forRoot(),
+    groupRouting],
+  exports: [GroupMainPageComponent, CreateGroupComponent, AddUsersComponent],
+})
+export class GroupModule {}
+
+// exports: [GroupMainPageComponent, CreateGroupComponent],
```
Comment (joao-sd) [inline apps/toboggan-app/src/app/group/group.module.ts:28]: Please remove left over

**Review by joao-sd — CHANGES_REQUESTED** [review]
Great PR so far! Very well done
I left a couple of comments.

Could we also add a test add groups route to groups.http? It would be easier to trigger this manually if needed.

We also need to refactor our app controller and split it between an user.controller.ts and groups.controller.ts. We're nesting all logic for both providers there, but it will become confusing to maintain over time.

Thanks!

File: apps/toboggan-app/src/app/group/components/add-users/add-users.component.ts (hunk 2 above)
Comment (sada-ajith) [reply apps/toboggan-app/src/app/group/components/add-users/add-users.component.ts:50]: we put these as a placeholder for API integration

File: apps/toboggan-app/src/app/group/components/create-group/create-group.component.ts (hunk 3 above)
Comment (sada-ajith) [reply apps/toboggan-app/src/app/group/components/create-group/create-group.component.ts:69]: we put these as a placeholder for API integration

File: apps/toboggan-app/src/app/group/group.module.ts (hunk 4 above)
Comment (sada-ajith) [reply apps/toboggan-app/src/app/group/group.module.ts:28]: Sure will do

**Review by joao-sd — APPROVED** [review]
LGTM!

**Pull request merged**